lib/tcpip_stack_direct.ml

```diff
@@ -77,10 +77,10 @@ struct
   let ipv4 { ipv4; _ } = ipv4
 
   let listen_udpv4 t ~port callback =
-    Hashtbl.replace t.udpv4_listeners port callback
+    Hashtbl.replace t.udpv4_listeners (port mod 65536) callback
```

Comment:
I think using `mod 65536` is counterintuitive here: if a developer calls `listen_xxx` with a `port` greater than `2^16`, there's no way to find out on which port the service actually listens.

instead, I'd document the 16 bit size of port clearly and throw an exception if violated.